doc/adr/0011-concept-exception-handling-backend.md

```diff
@@ -0,0 +1,33 @@
+# 10. How to use exceptions in the backend
+
+Date: 2024-06-28
+
+## Status
+
+Accepted
+
+## Context
+So far we used different approaches when to use Optionals, throw checked exceptions or unchecked exceptions. This also led
+to using exceptions from other layers. We want to streamline our approach.
+
+## Decision
+Layer-Specific Exceptions
+* By having each layer (ports, adapters, and application) define its own exceptions, we achieve a clear separation of concerns. This ensures that exceptions are relevant to the specific context in which they occur.
+* Avoiding a common exception package prevents mixing unrelated exceptions and promotes a more organized codebase.
+
+Custom Runtime Exceptions
+* Throwing custom runtime exceptions in the application layer allows us to encapsulate domain-specific logic. These exceptions can be tailored to the norms project’s requirements.
+* Using @ControllerAdvice to handle these exceptions ensures consistent error handling across the application. It centralizes exception handling logic and provides a single point of control.
```

Comment:
I would maybe add here the main argument for this approach and that is that our backend app is a merely API supporting app, so that we can centralized the handling of the exceptions in the controller layer